### PR TITLE
fix: [cherry-pick] fix datarace in root_coord proxy_client_manager

### PR DESCRIPTION
**Pull request by wayblink**
pr: #28796 
issue: #28797

### CODE REVIEW

File: internal/rootcoord/root_coord.go [hunk 1]
```diff
@@ -2785,10 +2785,8 @@ func (c *Core) CheckHealth(ctx context.Context, in *milvuspb.CheckHealthRequest)
 
 	mu := &sync.Mutex{}
 	group, ctx := errgroup.WithContext(ctx)
-	errReasons := make([]string, 0, len(c.proxyClientManager.proxyClient))
-
-	c.proxyClientManager.lock.RLock()
-	for nodeID, proxyClient := range c.proxyClientManager.proxyClient {
+	errReasons := make([]string, 0, c.proxyClientManager.GetProxyCount())
+	for nodeID, proxyClient := range c.proxyClientManager.GetProxyClients() {
```
Comment (czs007) [inline internal/rootcoord/root_coord.go:2789]: Returning the underlying map object directly here does not guarantee concurrent safety.

Comment (xiaofan-luan) [reply internal/rootcoord/root_coord.go:2789]: Agreed, we have use sync map or copy

**Pull request closed without merging**